component/prometheus/fanout.go

```diff
@@ -23,14 +26,21 @@ type Fanout struct {
 	// children is where to fan out.
 	children []storage.Appendable
 	// ComponentID is what component this belongs to.
-	componentID string
+	componentID  string
+	writeLatency prometheus.Histogram
 }
 
 // NewFanout creates a fanout appendable.
-func NewFanout(children []storage.Appendable, componentID string) *Fanout {
+func NewFanout(children []storage.Appendable, componentID string, register prometheus.Registerer) *Fanout {
+	wl := prometheus.NewHistogram(prometheus.HistogramOpts{
+		Name: "agent_prometheus_fanout_latency",
+		Help: "Write latency for fanout to write to child tree",
```

Comment:
Should "child tree" be "children?"

Reply:
I wanted to indicate that it's the whole tree, not just immediate children. Happy to change but worry that detail will get lost.

Reply:
I think rephrasing it might be helpful, what about saying something like "direct and indirect components"